desktop/core/src/desktop/js/reactComponents/TaskBrowser/TaskBrowser.tsx

```diff
@@ -0,0 +1,434 @@
+                return new Date(b.date_done) - new Date(a.date_done);
+            });
+            setTasks(sortedTasks);
+          } else {
+            console.error('Expected an array of tasks, but received:', response.data);
+          }
+        })
+        .catch(error => {
+            console.error('Error fetching tasks', error);
+        });
+    };
+
+    fetchTasks();   //fetch tasks initially
+    const interval = setInterval(fetchTasks, 5000);
+    return () => clearInterval(interval); // Cleanup on unmount
+
+  }, []);
+
+  const handleTaskIdClick = (taskId) => {
+    setCurrentTaskId(taskId);
+    setShowLogsPopup(true);
+    // Here, you could also fetch the logs immediately if you want to
+    fetchTaskLogs(taskId);
+  };
+
+
+
+  const handleSearchChange = (e) => {
+   console.log('Filtering tasks by:', e.target.value);
+    setSearchTerm(e.target.value.toLowerCase());
+    };
+
+  // Function to handle status filter button clicks
+  const handleStatusFilterChange = (status) => {
+    setStatusFilter(prevStatusFilter => {
+      const isAll = status === 'all';
+      const newStatusFilter = {
+        ...prevStatusFilter,
+        [status]: isAll ? true : !prevStatusFilter[status],
+      };
+  
+      if (isAll) {
+        // If 'all' is selected, set everything else to false
+        newStatusFilter.success = false;
+        newStatusFilter.failure = false;
+        newStatusFilter.running = false;
+      } else {
+        // If any specific status is toggled, set 'all' to false
+        newStatusFilter.all = false;
+      }
+  
+      // If no individual statuses are selected, default to 'all'
+      if (!newStatusFilter.success && !newStatusFilter.failure && !newStatusFilter.running) {
+        newStatusFilter.all = true;
+      }
+  
+      return newStatusFilter;
+    });
+  };
+  
+  
+
+  const filteredTasks = tasks.filter(task => {
+    const taskNameMatch = task.result?.task_name?.toLowerCase().includes(searchTerm);
+    const userIdMatch = task.result?.username?.toString().toLowerCase().includes(searchTerm);
+    const taskIdMatch = task.task_id?.toString().toLowerCase().includes(searchTerm);
+    // const statusMatch = statusFilter === 'all' || task.status.toLowerCase() === statusFilter;
+    const statusMatch = statusFilter.all || 
+                      (statusFilter.success && task.status.toLowerCase() === 'success') ||
+                      (statusFilter.failure && task.status.toLowerCase() === 'failure') ||
+                      (statusFilter.running && task.status.toLowerCase() === 'running');
+
+
+
+    return (taskNameMatch || userIdMatch || taskIdMatch) && statusMatch;
+  });
+
+  const handleKillTask = (taskId) => {
+    axios.post(`/desktop/api2/kill_task/${taskId}/`)
+      .then(response => {
+        const { status, message } = response.data;
+        if (status === 'success') {
+          alert(`Task: ${taskId} has been killed.`);
```

Comment:
We don't use alert. You could consider `huePubSub.publish('hue.global.info', { message: ... })`

Reply:
changed it to `huePubSub.publish('hue.global.info', { message: ... })`